base/linalg/diagonal.jl

```diff
@@ -168,9 +168,9 @@ end
 
 A_mul_Bt(D::Diagonal, B::AbstractTriangular) = A_mul_B!(D, transpose(B))
 function A_mul_Bt(D::Diagonal, A::AbstractMatrix)
-    Ac = similar(A, promote_op(*, eltype(A), eltype(D.diag)), (size(A, 2), size(A, 1)))
-    ctranspose!(Ac, A)
-    A_mul_B!(D, Ac)
+    At = similar(A, promote_op(*, eltype(A), eltype(D.diag)), (size(A, 2), size(A, 1)))
```

Comment:
while you're adjusting this, I think it make more sense if the name of the second input were `B` rather than `A`